### PR TITLE
[TF FE]: Support complex tensors for ScatterNd operation

### PR DESCRIPTION
**Pull request by hub-bla**
### Details:
 - Support complex tensors for ScatterNd operation + tests

### Tickets:
 - [None](https://github.com/openvinotoolkit/openvino/issues/23240)

### CODE REVIEW

File: tests/layer_tests/tensorflow_tests/test_tf_ScatterND.py [hunk 1]
```diff
@@ -73,3 +75,56 @@ def test_tf_scatter_nd(self, params, ie_device, precision, ir_version, temp_dir,
...
+
+        return tf_net, None
+
+    test_data = [
+        dict(x_shape=[8], indices=[[4], [3], [1], [7]], updates_shape=[4]),
+        dict(x_shape=[10], indices=[[0], [2], [4], [6], [8]], updates_shape=[5]),
+        dict(x_shape=[5, 5], indices=[[0, 0], [1, 1], [2, 2], [3, 3]], updates_shape=[4]),
+        dict(x_shape=[3, 3, 3], indices=[[0, 0, 0], [1, 1, 1], [2, 2, 2]], updates_shape=[3]),
+    ]
+
```
Comment (rkazants) [inline tests/layer_tests/tensorflow_tests/test_tf_ScatterND.py:119]: please add one more parameter `indices_type` that can be int16, int32, int64. `shape` is of the same type as `indices`

**Review by rkazants — COMMENTED** [review]
left one comment. The remained part looks good to me. Thanks

**Comment by rkazants** [comment]
build_jenkins

File: tests/layer_tests/tensorflow_tests/test_tf_ScatterND.py (hunk 1 above)
Comment (hub-bla) [reply tests/layer_tests/tensorflow_tests/test_tf_ScatterND.py:119]: @rkazants So, I did tried to add `indices_type` for types mentioned above but it seems like there is a problem with the int16 type.

```
try:
    c_op = pywrap_tf_session.TF_FinishOperation(op_desc)
except errors.InvalidArgumentError as e:
    # Convert to ValueError for backwards compatibility.
>   raise ValueError(e.message)

E   ValueError: Input tensor must be int32 or int64, but was int16

ov-py310/lib/python3.10/site-packages/tensorflow/python/framework/ops.py:1020: ValueError
```

Let me know if I should skip it then.

Comment (rkazants) [reply tests/layer_tests/tensorflow_tests/test_tf_ScatterND.py:119]: if this TF error, problem looks in documentation or in TF implementation. So let us skip int16 then.

File: tests/layer_tests/tensorflow_tests/test_tf_ScatterND.py [hunk 2]
```diff
@@ -73,3 +75,56 @@ def test_tf_scatter_nd(self, params, ie_device, precision, ir_version, temp_dir,
...
+        dict(x_shape=[10], indices=[[0], [2], [4], [6], [8]], updates_shape=[5], indices_type=np.int64),
+        dict(x_shape=[5, 5], indices=[[0, 0], [1, 1], [2, 2], [3, 3]], updates_shape=[4], indices_type=np.int64),
+        dict(x_shape=[3, 3, 3], indices=[[0, 0, 0], [1, 1, 1], [2, 2, 2]], updates_shape=[3], indices_type=np.int32),
+    ]
+
+    @pytest.mark.parametrize("params", test_data)
+    @pytest.mark.nightly
+    @pytest.mark.precommit
+    @pytest.mark.xfail(condition=platform.system() == 'Darwin' and platform.machine() == 'arm64',
+                       reason='Ticket - 122716')
```
Comment (rkazants) [inline tests/layer_tests/tensorflow_tests/test_tf_ScatterND.py:124]:
```suggestion
```

**Review by rkazants — APPROVED** [review]

**Comment by rkazants** [comment]
build_jenkins

**Pull request merged**